contracts/native/version-control/src/commands.rs

```diff
@@ -198,6 +198,36 @@ pub fn yank_module(deps: DepsMut, msg_info: MessageInfo, module: ModuleInfo) ->
     ))
 }
 
+/// Set a module monetization allowing the namespace owner to charge for module installation/usage or else.
+pub fn set_module_monetization(
+    deps: DepsMut,
+    msg_info: MessageInfo,
+    module: ModuleInfo,
+    monetization: Monetization,
+) -> VCResult {
+    // validate the caller is the owner of the namespace
+
+    if module.namespace == Namespace::unchecked(ABSTRACT_NAMESPACE) {
+        // Only Admin can update abstract contracts
+        cw_ownable::assert_owner(deps.storage, &msg_info.sender)?;
+    } else {
+        // Only owner can add modules
+        validate_account_owner(deps.as_ref(), &module.namespace, &msg_info.sender)?;
+    }
+
+    // We verify the module exists before updating the monetization
+    REGISTERED_MODULES
```

Comment:
The `Module` struct also contains a version, which is not actually used by the store. Hence when adding some monetization we should either:
- accept only the module id and not the Module object
- or change this line to range over the map with `.take(1)` to check that there is at least one module registered. That will essentially ignore the version field of the provided Module (which might be confusing to a user).

Reply:
I'd say we accept only the module id + namespace, and not the whole module object. 
However this solution would complicate the different types that we need to provide to different endpoints. What do you think ?